apollo-router/src/query_planner/execution.rs

```diff
@@ -306,16 +307,33 @@ impl PlanNode {
                                     .current_dir(current_dir.clone())
                                     .deferred_fetches(parameters.deferred_fetches.clone())
                                     .build();
-                                let (v, e) = service
-                                    .oneshot(request)
-                                    .instrument(tracing::info_span!(
+                                let connector = parameters
+                                    .schema
+                                    .connectors_by_service_name
+                                    .as_ref()
+                                    .and_then(|connectors| {
+                                        connectors.get(fetch_node.service_name.as_str())
+                                    });
+                                let span = if let Some(connector) = connector {
+                                    tracing::info_span!(
+                                        CONNECT_SPAN_NAME,
+                                        "otel.kind" = "INTERNAL",
+                                        "apollo.connector.field" =
+                                            connector.id.directive.field.to_string(), // TODO: add field type and "!" if required
+                                        "apollo.connector.type" = "http", // TODO: fix after preview when more types are added
+                                        "apollo.connector.detail" = connector.id.label.as_str(),
```

Comment:
cc @BrynCooke this might need your attention

Reply:
Moving to Draft as there is still some ongoing discussion around these attributes. I'll add Bryn to the doc.